cimi-resources/src/com/sixsq/slipstream/ssclj/resources/storage_bucket.clj

```diff
@@ -0,0 +1,106 @@
+(ns com.sixsq.slipstream.ssclj.resources.storage-bucket
+  (:require
+    [clojure.string :as str]
+    [com.sixsq.slipstream.auth.acl :as a]
+    [com.sixsq.slipstream.ssclj.resources.common.crud :as crud]
+    [com.sixsq.slipstream.ssclj.resources.common.schema :as c]
+    [com.sixsq.slipstream.ssclj.resources.common.std-crud :as std-crud]
+    [com.sixsq.slipstream.ssclj.resources.common.utils :as u]
+    [clojure.pprint :as pp]
+    [com.sixsq.slipstream.ssclj.resources.spec.storage-bucket :as bucky]))
+
+(def ^:const resource-tag :storageBuckets)
+
+(def ^:const resource-name "StorageBucket")
+
+(def ^:const resource-url (u/de-camelcase resource-name))
+
+(def ^:const collection-name "StorageBucketCollection")
+
+(def ^:const resource-uri (str c/slipstream-schema-uri resource-name))
+
+(def ^:const collection-uri (str c/slipstream-schema-uri collection-name))
+
+;; only authenticated users can view and create credentials
+(def collection-acl {:owner {:principal "ADMIN"
+                             :type      "ROLE"}
+                     :rules [{:principal "ADMIN"
+                              :type      "ROLE"
+                              :right     "MODIFY"}
+                             {:principal "USER"
+                              :type      "ROLE"
+                              :right     "VIEW"}]})
+
+;;
+;; multimethod for ACLs
+;;
+
+(defn create-acl [id]
+  {:owner {:principal "ADMIN"
+           :type      "ROLE"}
+   :rules [{:principal id
+            :type      "USER"
+            :right     "VIEW"}]})
+
+;;
+;; set the resource identifier to "storage-bucket/uuid(connector-href, instanceID)"
+;;
+(defmethod crud/new-identifier resource-name [json resource-name]
+  (let [credential-href (get (first (get json :credentials)) :href)
+        bucketName (get json :bucketName)
```

Comment:
Neither of these values of very idiomatic.  These would be better replaced by:
```
(-> json :credentials first :href)
```
and
```
(:bucketName json)
```
The functions `get` and `get-in` are avoided in most cases.

Reply:
pushed